src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -1577,25 +1577,288 @@ class QuantizedOr(QuantizedOpUnivariateOfEncrypted, QuantizedOp):
     _impl_for_op_named: str = "Or"
 
 
-class QuantizedDiv(QuantizedOpUnivariateOfEncrypted, QuantizedOp):
-    """Div operator /.
+class QuantizedDiv(QuantizedMixingOp):
+    """Quantized Division operator.
 
-    This operation is not really working as a quantized operation. It just works when things got
-    fused, as in e.g., Act(x) = 1000 / (x + 42))
+    Can divide either two variables (both encrypted) or a variable and a constant
     """
 
     _impl_for_op_named: str = "Div"
 
+    def __init__(
+        self,
+        *args,
+        rounding_threshold_bits: Union[None, int, Dict[str, Union[str, int]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, rounding_threshold_bits=rounding_threshold_bits, **kwargs)
+        self.divider_quantizer: UniformQuantizer
+        self.min_non_zero_value: numpy.float64
+
+    def calibrate(self, *inputs: numpy.ndarray) -> numpy.ndarray:
+        """Create corresponding QuantizedArray for the output of the activation function.
+
+        Args:
+            *inputs (numpy.ndarray): Calibration sample inputs.
+
+        Returns:
+            numpy.ndarray: the output values for the provided calibration samples.
+        """
+
+        # If two inputs, compute the divider
+        if not self.can_fuse() and len(inputs) == 2:
+            min_non_zero_value = numpy.min(numpy.abs(inputs[1]))
+
+            # mypy
+            assert min_non_zero_value is not None and min_non_zero_value > 0
+            self.min_non_zero_value = min_non_zero_value
+
+            q_array_divider = QuantizedArray(self.n_bits, 1 / inputs[1])
```

Comment:
also, how can we be sure that `inputs[1]`is not 0 ? I feel like `1 / inputs[1]` can fail here no ?

Reply:
I do this right before:

```python

Reply:
ah yes right, I missed the assert. But does that mean that we always expect values to be strictly positive ? how can we ensure that ?

Reply:
What I could check is that no value are quant dequant to a 0 maybe

Reply:
yes but my question is more about "how can you be sure of that ?" !

Reply:
I am sorry I have lost the context here.

> But does that mean that we always expect values to be strictly positive ?

We check the min abs (`numpy.min(numpy.abs(inputs[1]))`). Not sure why the strictly positive.

Reply:
you are right we do the abs, so my questions is then : how can we be sure that we never get 0 in `inputs[1]`  ? I understand that we assert or could add more assert, but I don't get how we are sure that this/these assert(s) will never ever fail in any circumstances (and if we can't, that's an issue imo) !

Reply:
We can't. If the user provides a 0 it will fail. As any division fails when there are 0 in the denominator.

Reply:
ah I see, in that case if it's a user issue, then we should have a `raise ValueError` with an explicit message rather than an empty assert. Unless an error is triggered before reaching this point (and in that case, is it explicit enough ?)

Reply:
The assert here is for mypy. Not sure we should rewrite basic error message such as division by 0.

Reply:
well I won't push too much on this one, but I just fear that the error will not be a clear "DivisionByZeroError" and might confuse the user, would be worth a check or even a test